website/addons/googledrive/model.py

```diff
@@ -2,235 +2,189 @@
 """Persistence layer for the google drive addon.
 """
 import os
-from urllib import unquote
-from datetime import datetime
+import urllib
 
+from datetime import datetime
 from modularodm import fields
 
 from framework.auth import Auth
-from framework.mongo import StoredObject
+from website.oauth.models import ExternalProvider
 
 from website.addons.base import exceptions
-from website.addons.base import AddonUserSettingsBase, AddonNodeSettingsBase
 from website.addons.base import StorageAddonBase
+from website.addons.base import AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase
 
-from website.addons.googledrive.client import GoogleAuthClient
 from website.addons.googledrive import settings as drive_settings
-from website.addons.googledrive.utils import GoogleDriveNodeLogger
+from website.addons.googledrive.serializer import GoogleDriveSerializer
+from website.addons.googledrive.client import GoogleAuthClient, GoogleDriveClient
 
 
-class GoogleDriveOAuthSettings(StoredObject):
-    """
-    this model address the problem if we have two osf user link
-    to the same google drive user and their access token conflicts issue
-    """
+class GoogleDriveProvider(ExternalProvider):
+    name = 'Google Drive'
+    short_name = 'googledrive'
 
-    # google drive user id, for example, "4974056"
-    user_id = fields.StringField(primary=True, required=True)
-    # google drive user name this is the user's login
-    username = fields.StringField()
-    access_token = fields.StringField()
-    refresh_token = fields.StringField()
-    expires_at = fields.DateTimeField()
-
-    def fetch_access_token(self, force_refresh=False):
-        self.refresh_access_token(force=force_refresh)
-        return self.access_token
+    client_id = drive_settings.CLIENT_ID
+    client_secret = drive_settings.CLIENT_SECRET
 
-    def refresh_access_token(self, force=False):
-        if self._needs_refresh() or force:
-            client = GoogleAuthClient()
-            token = client.refresh(self.access_token, self.refresh_token)
+    auth_url_base = "https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force"
+    callback_url = 'https://www.googleapis.com/oauth2/v3/token'
```

Comment:
The overall goal is to define reused settings variables in a consistent standardized place for all usages; there's some flexibility in how you go about it.

Try to avoid string concatenation though; it's a bit icky. It's worth noting that `_build_url` seems to be a needlessly overcomplicated wrapper for `furl`.